From 955a042c01df428b88d0079d112e33df92246bcf Mon Sep 17 00:00:00 2001
From: Alain Volmat <alain.volmat@foss.st.com>
Date: Thu, 25 Jan 2024 09:36:22 +0100
Subject: [PATCH 0839/1141] media: dcmipp: byteproc: correct bpp for RAW format
 > RAW8

Correct error of bpp in case of RAW formats larger than RAW8.
RAW10, RAW12, RAW14 and all unpacked to 16bit.

Change-Id: I1dbb7059a3cb10c4b9ae48490f7b775d4bf1f23f
Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/354230
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Philippe CORNU <philippe.cornu@foss.st.com>
Reviewed-by: Philippe CORNU <philippe.cornu@foss.st.com>
Reviewed-by: Hugues FRUCHET <hugues.fruchet@st.com>
---
 .../st/stm32/stm32-dcmipp/dcmipp-byteproc.c   | 24 +++++++++----------
 1 file changed, 12 insertions(+), 12 deletions(-)

--- a/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-byteproc.c
+++ b/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-byteproc.c
@@ -72,18 +72,18 @@ static const struct dcmipp_byteproc_pix_
 	PIXMAP_MBUS_BPP(SGBRG8_1X8, 1, MIPI_CSI2_DT_RAW8),
 	PIXMAP_MBUS_BPP(SGRBG8_1X8, 1, MIPI_CSI2_DT_RAW8),
 	PIXMAP_MBUS_BPP(SRGGB8_1X8, 1, MIPI_CSI2_DT_RAW8),
-	PIXMAP_MBUS_BPP(SBGGR10_1X10, 1, MIPI_CSI2_DT_RAW10),
-	PIXMAP_MBUS_BPP(SGBRG10_1X10, 1, MIPI_CSI2_DT_RAW10),
-	PIXMAP_MBUS_BPP(SGRBG10_1X10, 1, MIPI_CSI2_DT_RAW10),
-	PIXMAP_MBUS_BPP(SRGGB10_1X10, 1, MIPI_CSI2_DT_RAW10),
-	PIXMAP_MBUS_BPP(SBGGR12_1X12, 1, MIPI_CSI2_DT_RAW12),
-	PIXMAP_MBUS_BPP(SGBRG12_1X12, 1, MIPI_CSI2_DT_RAW12),
-	PIXMAP_MBUS_BPP(SGRBG12_1X12, 1, MIPI_CSI2_DT_RAW12),
-	PIXMAP_MBUS_BPP(SRGGB12_1X12, 1, MIPI_CSI2_DT_RAW12),
-	PIXMAP_MBUS_BPP(SBGGR14_1X14, 1, MIPI_CSI2_DT_RAW14),
-	PIXMAP_MBUS_BPP(SGBRG14_1X14, 1, MIPI_CSI2_DT_RAW14),
-	PIXMAP_MBUS_BPP(SGRBG14_1X14, 1, MIPI_CSI2_DT_RAW14),
-	PIXMAP_MBUS_BPP(SRGGB14_1X14, 1, MIPI_CSI2_DT_RAW14),
+	PIXMAP_MBUS_BPP(SBGGR10_1X10, 2, MIPI_CSI2_DT_RAW10),
+	PIXMAP_MBUS_BPP(SGBRG10_1X10, 2, MIPI_CSI2_DT_RAW10),
+	PIXMAP_MBUS_BPP(SGRBG10_1X10, 2, MIPI_CSI2_DT_RAW10),
+	PIXMAP_MBUS_BPP(SRGGB10_1X10, 2, MIPI_CSI2_DT_RAW10),
+	PIXMAP_MBUS_BPP(SBGGR12_1X12, 2, MIPI_CSI2_DT_RAW12),
+	PIXMAP_MBUS_BPP(SGBRG12_1X12, 2, MIPI_CSI2_DT_RAW12),
+	PIXMAP_MBUS_BPP(SGRBG12_1X12, 2, MIPI_CSI2_DT_RAW12),
+	PIXMAP_MBUS_BPP(SRGGB12_1X12, 2, MIPI_CSI2_DT_RAW12),
+	PIXMAP_MBUS_BPP(SBGGR14_1X14, 2, MIPI_CSI2_DT_RAW14),
+	PIXMAP_MBUS_BPP(SGBRG14_1X14, 2, MIPI_CSI2_DT_RAW14),
+	PIXMAP_MBUS_BPP(SGRBG14_1X14, 2, MIPI_CSI2_DT_RAW14),
+	PIXMAP_MBUS_BPP(SRGGB14_1X14, 2, MIPI_CSI2_DT_RAW14),
 	PIXMAP_MBUS_BPP(JPEG_1X8, 1, 0x00), /* TODO - DT value to be fixed */
 };
 
